great_expectations/data_context/store/store.py

```diff
@@ -20,6 +22,12 @@
 logger = logging.getLogger(__name__)
 
 
+class StoreConfigTypedDict(TypedDict):
+    class_name: str
+    module_name: str
+    store_backend: dict
```

Comment:
I can drop these typing changes if they are not useful.

Reply:
are we always guaranteed these values? or these are just possible keys?

Reply:
They are not guaranteed, unfortunately.
I created this list by just creating this type and then adding fields based on the typing errors `mypy` gave me.

For example based on errors users are reporting we know that `module_name` is not always defined. What I'm not sure about is... should `module_name` always be present? In which case this `TypeDict` could help us enforce this behavior and or should we expect the `module_name` to sometimes not be present.

I'm going to add some comments and look at some tests before merging to help answer these questions.